mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraACLMapper.java

```diff
@@ -19,66 +19,169 @@
+            return cassandraACLDAOV1.setACL(cassandraId, mailboxACL)
+                .flatMap(aclDiff -> userMailboxRightsDAO.update(cassandraId, aclDiff)
+                    .thenReturn(aclDiff))
+                .switchIfEmpty(Mono.defer(() -> Mono.error(new MailboxException("Unable to update ACL"))));
+        }
+
+        public Mono<Void> delete(CassandraId cassandraId) {
+            return cassandraACLDAOV1.getACL(cassandraId)
+                .flatMap(acl -> userMailboxRightsDAO.update(cassandraId, ACLDiff.computeDiff(acl, MailboxACL.EMPTY))
+                    .then(cassandraACLDAOV1.delete(cassandraId)));
+        }
+    }
+
+    public static class StoreV2 implements Store {
+        private final CassandraACLDAOV2 cassandraACLDAOV2;
+        private final EventSourcingSystem eventSourcingSystem;
+
+        @Inject
+        public StoreV2(CassandraUserMailboxRightsDAO userMailboxRightsDAO,
+                CassandraACLDAOV2 cassandraACLDAOV2,
+                EventStore eventStore) {
+            this.cassandraACLDAOV2 = cassandraACLDAOV2;
+            Set<CommandHandler<? extends Command>> commandHandlers = ImmutableSet.of(new DeleteMailboxCommand.CommandHandler(eventStore),
+                new UpdateACLCommand.CommandHandler(eventStore),
+                new SetACLCommand.CommandHandler(eventStore));
+            Set<Subscriber> subscribers = ImmutableSet.of(new UserRightsDAOSubscriber(userMailboxRightsDAO),
+                new AclV2DAOSubscriber(cassandraACLDAOV2));
+            eventSourcingSystem = EventSourcingSystem.fromJava(commandHandlers, subscribers, eventStore);
+        }
+
+        @Override
+        public Mono<MailboxACL> getACL(CassandraId cassandraId) {
+            return cassandraACLDAOV2.getACL(cassandraId);
+        }
+
+        @Override
+        public Mono<ACLDiff> updateACL(CassandraId cassandraId, MailboxACL.ACLCommand command) {
+            return Mono.from(eventSourcingSystem.dispatch(new UpdateACLCommand(new MailboxAggregateId(cassandraId), command)))
+                .flatMapIterable(events -> events)
+                .filter(ACLUpdated.class::isInstance)
+                .map(ACLUpdated.class::cast)
+                .map(ACLUpdated::getAclDiff)
+                .next()
+                .switchIfEmpty(Mono.defer(() -> Mono.error(new MailboxException("Unable to update ACL"))));
+        }
+
+        @Override
+        public Mono<ACLDiff> setACL(CassandraId cassandraId, MailboxACL mailboxACL) {
+            return Mono.from(eventSourcingSystem.dispatch(new SetACLCommand(new MailboxAggregateId(cassandraId), mailboxACL)))
+                .flatMapIterable(events -> events)
+                .filter(ACLReseted.class::isInstance)
+                .map(ACLReseted.class::cast)
+                .map(ACLReseted::getAclDiff)
+                .next()
+                .switchIfEmpty(Mono.defer(() -> Mono.error(new MailboxException("Unable to update ACL"))));
```

Comment:
Maybe the log error should be `"Unable to update ACL"`, to differentiate with the one above?